homeassistant/components/sonos/sensor.py

```diff
@@ -100,7 +101,7 @@ def native_value(self) -> int | None:
     @property
     def available(self) -> bool:
         """Return whether this device is available."""
-        return self.speaker.available and self.speaker.power_source
+        return cast(bool, self.speaker.available and self.speaker.power_source)
```

Comment:
`speaker.power_source` is defined as `str | None`, the condition doesn't return a true boolean. Better to check for `is not None` instead.
```py
return self.speaker.available and self.speaker.power_source is not None
```